wrappers/rust/cpp-client-telemetry-sys/src/lib.rs

```diff
@@ -0,0 +1,90 @@
+#![allow(non_upper_case_globals)]
+#![allow(non_camel_case_types)]
+#![allow(non_snake_case)]
+include!(concat!(env!("OUT_DIR"), "/bindings.rs"));
+
+use std::ffi::*;
+
+fn evt_api_call_wrapper(evt_context: Box<evt_context_t>) -> (evt_status_t, Box<evt_context_t>) {
+    let raw_pointer = Box::into_raw(evt_context);
+
+    let mut result: evt_status_t = -1;
+    unsafe {
+        result = evt_api_call_default(raw_pointer);
+    }
+
+    let out_context = unsafe { Box::from_raw(raw_pointer) };
+    (result, out_context)
+}
+
+pub fn evt_open(config: &CString) -> Option<evt_handle_t> {
+    let config_bytes = config.to_bytes_with_nul().to_vec();
+
+    let context: Box<evt_context_t> = Box::new(evt_context_t {
+        call: evt_call_t_EVT_OP_OPEN,
+        handle: 0,
+        data: config_bytes.as_ptr() as *mut c_void,
+        result: 0,
+        size: 0,
+    });
+
+    let (result, context) = evt_api_call_wrapper(context);
+
+    if result == -1 {
+        return Option::None;
+    }
+
+    return Some(context.handle.clone());
+}
+
+pub fn evt_close(handle: &evt_handle_t) -> evt_status_t {
+    let context: Box<evt_context_t> = Box::new(evt_context_t {
+        call: evt_call_t_EVT_OP_CLOSE,
+        handle: *handle,
+        data: std::ptr::null_mut(),
+        result: 0,
+        size: 0,
+    });
+
+    let (result, _) = evt_api_call_wrapper(context);
+    result
+}
+
+pub fn evt_upload(handle: &evt_handle_t) -> evt_status_t {
```

Comment:
Because you are directly exposing `evt_handle_t` outside your crate, and this type has unsafe pointers in it, these functions are actually unsafe to call.  They allow a safe caller to violate type safety.

You can fix this in one of several ways:
1. Don't expose these functions outside this crate.
2. Don't expose `evt_context_t`, and instead define an `EvtContext` wrapper type which contains `evt_context_t` but does not expose it outside of the crate.
3. Mark these functions `unsafe`.